internal/server/iigointernal/legislature.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/SOMAS2020/SOMAS2020/internal/common/shared"
+	"github.com/SOMAS2020/SOMAS2020/internal/common/voting"
+	"github.com/pkg/errors"
+)
+
+type legislature struct {
+	SpeakerID     shared.ClientID
+	budget        shared.Resources
+	judgeSalary   shared.Resources
+	ruleToVote    string
+	ballotBox     voting.BallotBox
+	votingResult  bool
+	clientSpeaker roles.Speaker
+}
+
+// returnJudgeSalary returns the salary to the common pool.
+func (l *legislature) returnJudgeSalary() shared.Resources {
+	x := l.judgeSalary
+	l.judgeSalary = 0
+	return x
+}
+
+// withdrawJudgeSalary withdraws the salary for the Judge from the common pool.
+func (l *legislature) withdrawJudgeSalary(gameState *gamestate.GameState) error {
+	var judgeSalary = shared.Resources(rules.VariableMap["judgeSalary"].Values[0])
+	var withdrawError = WithdrawFromCommonPool(judgeSalary, gameState)
+	if withdrawError != nil {
+		l.judgeSalary = judgeSalary
+	}
+	return withdrawError
+}
+
+// sendJudgeSalary sets the budget of the Judge.
+func (l *legislature) sendJudgeSalary() {
+	amount, _ := l.clientSpeaker.PayJudge()
+	judicialBranch.budget = amount
+}
+
+// Receive a rule to call a vote on
+func (l *legislature) setRuleToVote(r string) {
+	l.ruleToVote, _ = l.clientSpeaker.DecideAgenda(l.ruleToVote)
+	//TODO:Handle errors properly
+}
+
+//Asks islands to vote on a rule
+//Called by orchestration
+func (l *legislature) setVotingResult(clientIDs []shared.ClientID) {
+
+	ruleID, participatingIslands, err := l.clientSpeaker.DecideVote(l.ruleToVote, clientIDs)
+	if err != nil {
+		return
+	}
+	l.ballotBox = l.RunVote(ruleID, participatingIslands)
+	l.votingResult = l.ballotBox.CountVotesMajority()
+
+}
+
+//RunVote creates the voting object, returns votes by category (for, against) in BallotBox.
+//Passing in empty ruleID or empty clientIDs results in no vote occurring
+func (l *legislature) RunVote(ruleID string, clientIDs []shared.ClientID) voting.BallotBox {
+
+	if ruleID == "" || len(clientIDs) == 0 {
+		return voting.BallotBox{}
+	}
+	l.budget -= serviceCharge
+	ruleVote := voting.RuleVote{}
+
+	//TODO: check if rule is valid, otherwise return empty ballot, raise error?
+	ruleVote.SetRule(ruleID)
+
+	//TODO: intersection of islands alive and islands chosen to vote in case of client error
+	//TODO: check if remaining slice is >0, otherwise return empty ballot, raise error?
+	ruleVote.SetVotingIslands(clientIDs)
+
+	ruleVote.GatherBallots(iigoClients)
+	//TODO: log of vote occurring with ruleID, clientIDs
+	//TODO: log of clientIDs vs islandsAllowedToVote
+	//TODO: log of ruleID vs s.RuleToVote
+	return ruleVote.GetBallotBox()
+}
+
+//Speaker declares a result of a vote (see spec to see conditions on what this means for a rule-abiding speaker)
+//Called by orchestration
+func (l *legislature) announceVotingResult() error {
+
+	rule, result, err := l.clientSpeaker.DecideAnnouncement(l.ruleToVote, l.votingResult)
+
+	if err == nil {
```

Comment:
What happens if err is not nil? 

Also this function has an error return type, but you aren't returning any errors.

Reply:
Reviewed and changed